pkgdb/src/resolver/manifest-raw.cc

```diff
@@ -462,6 +462,81 @@ to_json( nlohmann::json & jto, const EnvBaseRaw & env )
 }
 
 
+/* -------------------------------------------------------------------------- */
+
+static void
+from_json( const nlohmann::json & jfrom, ProfileScriptsRaw & profile )
+{
+  assertIsJSONObject<InvalidManifestFileException>(
+    jfrom,
+    "manifest field 'profile'" );
+
+  /* Clear fields */
+  profile.common = std::nullopt;
+  profile.bash   = std::nullopt;
+  profile.zsh    = std::nullopt;
+
+  /* Iterate over keys of the JSON object */
+  for ( const auto & [key, value] : jfrom.items() )
+    {
+      if ( key == "common" )
+        {
+          try
+            {
+              value.get_to( profile.common );
+            }
+          catch ( const nlohmann::json::exception & )
+            {
+              throw InvalidManifestFileException(
+                "failed to parse manifest field 'profile.common' with value: "
+                + value.dump() );
+            }
+        }
+      else if ( key == "bash" )
+        {
+          try
+            {
+              value.get_to( profile.bash );
+            }
+          catch ( const nlohmann::json::exception & )
+            {
+              throw InvalidManifestFileException(
+                "failed to parse manifest field 'profile.bash' with value: "
+                + value.dump() );
+            }
+        }
+      else if ( key == "zsh" )
+        {
+          try
+            {
+              value.get_to( profile.zsh );
+            }
+          catch ( const nlohmann::json::exception & )
+            {
+              throw InvalidManifestFileException(
+                "failed to parse manifest field 'profile.zsh' with value: "
+                + value.dump() );
+            }
+        }
+      else
+        {
+          throw InvalidManifestFileException(
+            "unrecognized manifest field 'profile." + key + "'." );
```

Comment:
should this be a more specific error message (e.g. unsupported shell script or similar)?

Reply:
Update coming.